linkerd/core/src/main/scala/io/buoyant/linkerd/Linker.scala

```diff
@@ -68,89 +87,45 @@ object Linker {
       }
 
       val namerParams = Stack.Params.empty + param.Tracer(tracer)
-      val interpreter = mkNameInterpreter(namers.getOrElse(Nil), namerParams)
-
-      val params = namerParams + DstBindingFactory.Namer(interpreter)
+      val namersByPrefix = namers.getOrElse(Nil).reverse.map { namer =>
+        namer.prefix -> namer.newNamer(namerParams)
+      }
 
       // Router labels must not conflict
       routers.groupBy(_.label).foreach {
-        case (label, rts) =>
-          if (rts.size > 1) throw ConflictingLabels(label)
+        case (label, rts) if rts.size > 1 =>
+          throw ConflictingLabels(label)
+        case _ =>
       }
```

Comment:
tioli: I'm not a huge fan of `case _ =>`, although the only alternative I can think of is to use `collectFirst` instead of `foreach`. something like: 

```scala
routers
  .groupBy(_.label)
  .collectFirst { case (label, rts) if rts.size > 1 => ConflictingLabels(label) }
  .foreach(throw _)
```